src/registrar/views/domain.py

```diff
@@ -831,6 +857,8 @@ def _send_domain_invitation_email(self, email: str, requestor: User, add_success
                     "requestor_email": requestor_email,
                 },
             )
+            if add_success:
```

Comment:
Moved this into the try statement since we shouldn't be sending emails if we do run into Exceptions that aren't EmailSendingError - feel free to let me know if there was a reason we did this previously though!